packages/intrepid2/unit-test/Orientation/test_orientation_HEX_newBasis.hpp

```diff
@@ -0,0 +1,763 @@
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Kyungjoo Kim  (kyukim@sandia.gov), or
+//                    Mauro Perego  (mperego@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+
+
+/** \file
+    \brief  Test for checking orientation tools for Hexahedral elements
+
+    The test considers two hexahedra in the physical space sharing a common face. 
+    In order to test significant configurations, we consider 6 mappings of the reference hexahedron
+    to the first (physical) hexahedron, so that the common face is mapped from all the 6 faces
+    of the reference hexahedron.
+    Then, for each of the mappings, the global ids of the vertices of the common face are permuted.
+    This gives a total of 144 combinations
+
+    The test considers HGRAD, HCURL and HDIV Lagrangian and Hierarchical basis functions, and for each of them:
+    1. Computes the oriented basis.
+    2. computes the basis coefficients, separately for each hexa, for functions belonging to the H-space spanned by the basis.
+    3. checks that the dofs shared between the two hexas are equivalent (this ensures that the orientation works correctly)
+    4. checks that the functions are indeed exactly reproduced
+
+    \author Created by Mauro Perego
+ */
+
+#include "Intrepid2_config.h"
+
+#ifdef HAVE_INTREPID2_DEBUG
+#define INTREPID2_TEST_FOR_DEBUG_ABORT_OVERRIDE_TO_CONTINUE
+#endif
+
+#include "Intrepid2_Orientation.hpp"
+#include "Intrepid2_OrientationTools.hpp"
+#include "Intrepid2_HGRAD_LINE_C1_FEM.hpp"
+#include "Intrepid2_HGRAD_QUAD_C1_FEM.hpp"
+#include "Intrepid2_HGRAD_HEX_C1_FEM.hpp"
+#include "Intrepid2_HGRAD_HEX_Cn_FEM.hpp"
+#include "Intrepid2_HCURL_HEX_In_FEM.hpp"
+#include "Intrepid2_HCURL_QUAD_In_FEM.hpp"
+#include "Intrepid2_HVOL_LINE_Cn_FEM.hpp"
+#include "Intrepid2_HVOL_QUAD_Cn_FEM.hpp"
+#include "Intrepid2_PointTools.hpp"
```

Comment:
I don't think you need any of these basis includes, and I think the code would be clearer if you deleted them.  Instead of these, you probably should include:
```
#include "Intrepid2_HierarchicalBasisFamily.hpp"
#include "Intrepid2_NodalBasisFamily.hpp"
```
which define the bases you are testing.